src/components/SelectMultiTypeahead.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { useState } from 'react';
+import type { MenuToggleElement } from '@patternfly/react-core/dist/js';
+import {
+  Button,
+  Chip,
+  ChipGroup,
+  MenuToggle,
+  Select,
+  SelectList,
+  SelectOption,
+  SelectOptionProps,
+  TextInputGroup,
+  TextInputGroupMain,
+  TextInputGroupUtilities,
+} from '@patternfly/react-core';
+import TimesIcon from '@patternfly/react-icons/dist/esm/icons/times-icon';
+
+const SelectMultiTypeahead = ({
+  options,
+  selected,
+  onChangeSelected,
+}: {
+  options: SelectOptionProps[];
+  selected: string[];
+  onChangeSelected: (newSelections: string[]) => void;
+}) => {
+  const [initialOptions] = useState(options);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [inputValue, setInputValue] = React.useState<string>('');
+  const [selectOptions, setSelectOptions] =
+    React.useState<SelectOptionProps[]>(options);
+  const [focusedItemIndex, setFocusedItemIndex] = React.useState<number | null>(
+    null
+  );
+  const [activeItem, setActiveItem] = React.useState<string | null>(null);
+  const textInputRef = React.useRef<HTMLInputElement>();
+  const elementId = `multi-typeahead-select`;
+
+  React.useEffect(() => {
+    let newSelectOptions: SelectOptionProps[] = initialOptions;
+
+    // Filter menu items based on the text input value when one exists
+    if (inputValue) {
+      newSelectOptions = initialOptions.filter((menuItem) =>
+        String(menuItem.children)
```

Comment:
Can we not iterate over children nodes but the options themselves?

Reply:
Filtering based on the `value` seems confusing. For instance, that wouldn't find `Identity & Access Management (iam)` for `id`.